### DISCUSSION TITLE
the Nachtzuster fork in a VM on proxmox - currently on the MatthewBCooke fork

### DISCUSSION THREAD

**Discussion by huytman** [Nachtzuster/BirdNET-Pi#55] [Q&A]
Hi all. Looks like this Nachtzuster fork of BirdNET-Pi is the one to use at the moment. I was delighted when I was able to set up BirdNET-Pi in a Debian container on my server running proxmox using the MatthewBCooke fork and instructions. I've tried more or less duplicating the install with this fork and I'm pretty sure it's an ARM thing (could be and likely wrong). Regardless - I wasn't able to get things working, Anyone else running this fork in a VM on proxmox? I'm willing to test out anyone's ideas. Thanks.

**Comment by lloydbayley** [comment]
I have been running mine for quite some time and migrated to Nachzusters fork without any problems.
When I originally installed mine on a VM, I had to make a custom CPU-type in ProxMox. I'll have to check if that is still required but if it is, I'm happy to share how.

**Reply by Nachtzuster** [reply]
Since you got it working with proxmox without any changes to BirdNET-Pi (I think):
What exactly isn't working without the custom CPU type?
I presume the host is a x86_64 machine?

**Reply by lloydbayley** [reply]
Jesus, I'm sorry, I didn't see this through all the static. The host is indeed an x86_64 machine.
The cpu was needed otherwise, it didn't run any detections. Nothing in the logs.
I must admit that I didn't pursue why not, I just accepted it and moved on.

**Reply by Nachtzuster** [reply]
Thanks - any idea if you are using the normal tflite_runtime as installed by the installer or manually installing tensorflow instead?

**Reply by lloydbayley** [reply]
It was installed exactly as you intended with the normal tflite_runtime. No deviation from the installer. I simply spun up a new VM, created the CPU type and installed BirdNet. Actually, it can be installed first and then the CPU can be added to make it work.

**Comment by lloydbayley** [comment]
Ok. I did a brand-spanker of a test install for you, to test this theory out. Indeed, without the custom CPU config, it doesn't show anything in the logs or attempt to process anything. After I applied the custom config, it was all up and running.

Here's what should work:

1) Log into the host Proxmox console, NOT the VM.

2) Create a new file or edit the existing.

```
nano /etc/pve/virtual-guest/cpu-models.conf
```

3) Add the following code:
```
cpu-model: BirdNet
    flags +sse4.1
    reported-model host
```

**Now go to the VM settings** 

4) Click on Hardware.

5) Double-click on Processors

6) In the Type field, scroll to the bottom and select the BirdNet processor we just created.

You will need to reboot the VM for the processor change to take effect.

Good luck! :)

**Reply by huytman** [reply]
Thank you Lloyd for checking into this. I'm stuck though. Seems like you're creating a VM and not a CT (container). Can you give a little background on the VM you're creating (OS used, # of CPUs, memory allotment, HD allotment)? Also - just to be clear - this is running on a x86 server running ProxMox? Again - thanks for the likely bright light at the end here - maybe just a few more tips until I get there.

**Reply by lloydbayley** [reply]
Well, to be fair, you DID ask about a VM, so that's why I responded about the VM.
The reason I used a VM was to make it easier to grab the whole thing in a snapshot backup and move to a new server (which I did just the other day and was dead-simple). I have a snapshot that runs in the wee small hours every day as a backup.

The test one I spun up was a small VM with not much ram or disk space but I'll give you the stats of my real working one.

OS: Debian GNU/Linux 12 (bookworm)
Kernel: Linux 6.5.13-1-pve
Model: Intel NUC12WSBi5
Type: 12th Gen Intel(R) Core(TM) i5-1240P

VM Details:

2 Cores
6GB RAM
300GB HDD

I don't think these are essential and less RAM (4GB?) might be fine etc...HDD size depends on how many entries/songfiles you wish to keep.
If you have a standard Purge Installation, then it won't go over-capacity, however, if you with to keep everything, start medium and you can increase as you need.

Hope that helps. Let me know if you have more questions.

**Comment by lloydbayley** [comment]
How did you go @huytman ? Have you had a chance to look at this yet?

**Comment by lloydbayley** [comment]
Checking in again. Any luck?

**Comment by huytman** [comment]
So sorry for the delayed response - I hadn't been able to give the time needed to try the things you mentioned until this evening. I set up a Debian 12 VM with specs you mentioned above.  Installed curl and attempt to run the install script from the main Nachtzuster page (curl..etc) and I get an error - 

sudo: a password is required
Passwordless sudo is not working. Aborting

Now - I accidentally used the same curl command but with the original birdnet-pi and it installed

Likely an obvious error on my part but any help would be appreciated. Also - I did create a user - pi

**Reply by lloydbayley** [reply]
I was beginning to think I was invisible! :)
That's a weird one. I suppose it really depends on how you did your Debian install.
Do you have the coreutils package installed? (I can't see why it wouldn't be but worth checking).
If not, just ```apt install coreutils``` and see how you go.
I really am at a loss as to why the original would install and this repo won't.

**Reply by Nachtzuster** [reply]
> Passwordless sudo is not working. Aborting

Assuming this is correct, I'm guessing you forgot to set up Passwordless sudo ?

**Reply by huytman** [reply]
That was it. I mentioned it was likely an obvious oversight on my part. I had not heard of passwordless sudo and somehow when I googled it initially it wasn't clear to me how to (or if I could) set it up. Done now and installing as expected. I will be sure to make your CPU change you first mentioned - and close out this line of questioning. Thanks!

**Reply by huytman** [reply]
made the CPU change and the GUI is up and running and 'live audio' is working - but no analysis. I get this error in the services menu under the 'analysis' service -

BirdNET Analysis (stalled - backlog of 251 files in ~/BirdSongs/StreamData/)

will continue to tinker - but if there's an obvious solution here - I would appreciate it.

**Reply by lloydbayley** [reply]
I don't understand why you would specifically need to set all this extra stuff up. It's a bit of a mystery. I just spun up a Debian VM and installed. Perhaps it's the way you installed Debian? It's an intriguing mystery. I always use the cloud-init-based installer as utilised by: https://helper-scripts.com/. Are you aware of that site? It's SERIOUSLY handy.

Anyway, as for the backlog problem, It could be quite a number of things. Have you looked at the files in ~/BirdSongs/StreamData/ ? Do they look correct?
It could error out if they weren't being written correctly as in too short (less than 3 seconds) or too long. Are you using a normal USB mic or an RTSP stream? 

Going to need a little more info!

Also, running

 ```bash ~/BirdNET-Pi/scripts/print_diagnostic_info.sh```

May yield some results of help.

**Comment by huytman** [comment]
I took your advice and used the helper scripts to set up a Debian 12 VM. I adjusted the memory to 8GB and the boot disk size to 100GB. I set up a 'pi' user and adjusted it to passwordless sudo. Installed with the curl command and all went fine. I changed CPU as suggested earlier. I am using an RTSP stream and can hear live audio through the BirdNET GUI. But - still getting the same error in the tool>services screen

**BirdNET Analysis (stalled - backlog of 326 files in ~/BirdSongs/StreamData/)**

I got back into the console on the VM and am seeing this error -

**[ 3317.622423] traps: python3[12286] trap invalid opcode ip:7ff8065bc354 sp:7fffed2f65c0 error:0 in _pywrap_tensorflow_interpreter_wrapper.so[7ff7f9a00000+f4a8000]**

Seems like a tensor flow issue. I will check into this more. As an FYI - I am running this on a Dell R720 using Intel Xeon CPU E5-2667 v2.

UPDATE 1: I tried all of the above with Debian 11 and more or less the same - installs fine - clicking on 'spectrogram' on the main menu plays live audio with the Spectrogram. Nothing is being analyzed though. I don't see the tensorflow error in the VM's console screen as with Debian 12. I had the MatthewBCooke fork of BirdNET running just fine in proxmox on the same server and BirdNET-Go in a docker inside an Ubuntu VM works for me. I'm thinking I should just stick with those.

**Comment by lloydbayley** [comment]
I'm sorry, when it comes to Tensorflow, I'm not going to be of much help. It's very odd that you're having so much trouble. I'm sorry I've failed here. As I said, it's a bit of a mystery. It's bound to be something simple but it's manifesting as something sneaky and complex. I'm struggling to see that it could be related to the CPU but again, who knows. Sorry about that.

**Comment by dwd1** [comment]
https://github.com/Nachtzuster/BirdNET-Pi/discussions/111

**Reply by alexbelgium** [reply]
You should add the title in addition to the number (not sure why sometimes github does it and somtimes not)

Post #111 : "Install BirdNET-Pi to Proxmox LXC container" (by dwd1)
